app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -206,8 +206,17 @@ def firmware_types
       {
         "automatic" => N_("Automatic"),
         "bios" => N_("BIOS"),
-        "efi" => N_("EFI"),
-      }
+        "uefi" => N_("UEFI"),
+        "uefi_secure_boot" => N_("UEFI Secure Boot"),
+      }.freeze
+    end
+
+    def firmware_type(vm_obj)
```

Comment:
```
return vm_obj.firmware unless vm_obj.firmware == 'efi'
vm_obj.secure_boot ? "uefi_secure_boot" : "uefi"
```

Reply:
Please no, I think that's distinctly worse to follow. The current flow is very easy to read IMHO.